src/components/ContactsList/index.js

```diff
@@ -33,31 +56,53 @@ export const withContacts = BaseComponent =>
     };
 
     async componentDidMount() {
-      const contactsRaw = await cozy.client
-        .fetchJSON("GET", "/data/io.cozy.contacts/_all_docs?include_docs=true")
-        .then(data => data.rows)
-        .catch(error => {
-          this.setState(state => ({ ...state, contacts: [], error }));
-        });
-      const contacts = (contactsRaw || [])
-        .map(row => ({
-          ...row.doc
-        }))
-        .filter(contact => contact.email);
+      const contacts = await cozyHTTP.getContacts();
       this.setState(state => ({
         ...state,
         contacts: [...state.contacts, ...contacts],
         loading: false
       }));
     }
 
+    deleteContact = async contact => {
+      const index = this.state.contacts.indexOf(contact);
+      this.setState(state => ({
+        ...state,
+        contacts: [
+          ...state.contacts.slice(0, index),
+          ...state.contacts.slice(index + 1)
+        ]
+      }));
```

Comment:
I delete contact from UI. But if something's wrong with the stack, I readd the contact. (see below)